### PR TITLE
builtin specprovider for locale nl

### PR DESCRIPTION
**Pull request by jorisdevrede**
Added the NLSpecProvider for the locale nl to generate a _burgerservicenummer_ or BSN. This is the Dutch equivalent of a social security number.

### CODE REVIEW

File: mimesis/builtins/nl.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        """Generate a random, but valid burgerservicenummer (BSN).
+
+        :returns: random BSN
+
+        :Example:
+            255159705
+        """
+
+        # a bsn is valid when its 9 digits pass the '11 test'
+        def is_valid_bsn(number):
```
Comment (lk-geimfari) [inline mimesis/builtins/nl.py]: This method must be private: `__is_valid_bsn`

File: mimesis/builtins/nl.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from mimesis.builtins.base import BaseSpecProvider
+
+
+class NLSpecProvider(BaseSpecProvider):
```
Comment (lk-geimfari) [inline mimesis/builtins/nl.py]: Maybe `NetherlandsSpecProvider`? It's much clearer for people which want import spec-provider for Netherlands.

**Comment by lk-geimfari** [comment]
@jorisdevrede I have two notes, but generally this is nice! Please, look at the review.

File: mimesis/builtins/nl.py (hunk 1 above)
Comment (jorisdevrede) [reply mimesis/builtins/nl.py]: You're right, although I believe this should be a single leading underscore `_is_valid_bsn` for non-public functions, right?

File: mimesis/builtins/nl.py (hunk 2 above)
Comment (jorisdevrede) [reply mimesis/builtins/nl.py]: Yes. I will change it.

File: mimesis/builtins/nl.py (hunk 1 above)
Comment (lk-geimfari) [reply mimesis/builtins/nl.py]: @jorisdevrede Absolutely.

File: mimesis/builtins/nl.py (hunk 2 above)
Comment (lk-geimfari) [reply mimesis/builtins/nl.py]: @jorisdevrede Okay! Thanks. Let me know, please.

File: mimesis/builtins/nl.py (hunk 1 above)
Comment (jorisdevrede) [reply mimesis/builtins/nl.py]: Done :-)

File: mimesis/builtins/nl.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        """Generate a random, but valid burgerservicenummer (BSN).
+
+        :returns: random BSN
+
+        :Example:
+            255159705
+        """
+
+        # a bsn is valid when its 9 digits pass the '11 test'
+        def _is_valid_bsn(number):
```
Comment (lk-geimfari) [inline mimesis/builtins/nl.py]: Annotation, please.

File: mimesis/builtins/nl.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+from mimesis.builtins.base import BaseSpecProvider
+
+
+class NetherlandsSpecProvider(BaseSpecProvider):
+    """Provides data specific to nl locale"""
+
+    class Meta:
+        name = 'nl_provider'
```
Comment (lk-geimfari) [inline mimesis/builtins/nl.py]: `netherlands_provider`

File: mimesis/builtins/nl.py (hunk 3 above)
Comment (jorisdevrede) [reply mimesis/builtins/nl.py]: Yes, of course. I've added annotations.

File: mimesis/builtins/nl.py (hunk 4 above)
Comment (jorisdevrede) [reply mimesis/builtins/nl.py]: Good catch! I changed it and added an additional test to cover it properly.

**Pull request merged**